client/src/main/java/com/alibaba/nacos/client/config/http/ServerHttpAgent.java

```diff
@@ -130,8 +132,8 @@
             
         } while (System.currentTimeMillis() <= endTime);
         
-        LOGGER.error("no available server");
-        throw new ConnectException("no available server");
+        LOGGER.error("Request time out or no available server, lastErrorSeverAddr: {}", lastErrorSeverAddr);
+        throw new ConnectException("Request time out or no available server, lastErrorSeverAddr: " + lastErrorSeverAddr);
```

Comment:
The last error server is no useful. When code run in there, means all server nodes has retried. only print last one can't help know the problem.

Reply:
Please check my comment below: 
```
    @Override
    public HttpRestResult<String> httpGet(String path, Map<String, String> headers, Map<String, String> paramValues,
            String encode, long readTimeoutMs) throws Exception {
        final long endTime = System.currentTimeMillis() + readTimeoutMs; // !!!!!
        injectSecurityInfo(paramValues);
        String currentServerAddr = serverListMgr.getCurrentServerAddr();
        int maxRetry = this.maxRetry;
        HttpClientConfig httpConfig = HttpClientConfig.builder()
                .setReadTimeOutMillis(Long.valueOf(readTimeoutMs).intValue())   // set read timeout here
                .setConTimeOutMillis(ConfigHttpClientManager.getInstance().getConnectTimeoutOrDefault(100)).build();
        do {
            ... ...
                HttpRestResult<String> result = NACOS_RESTTEMPLATE
                        .get(getUrl(currentServerAddr, path), httpConfig, newHeaders, query, String.class);
            ... ...
            } catch (ConnectException connectException) {
                LOGGER.error("[NACOS ConnectException httpGet] currentServerAddr:{}, err : {}",
                        serverListMgr.getCurrentServerAddr(), connectException.getMessage());
            } catch (SocketTimeoutException socketTimeoutException) {
                LOGGER.error("[NACOS SocketTimeoutException httpGet] currentServerAddr:{}， err : {}",
                        serverListMgr.getCurrentServerAddr(), socketTimeoutException.getMessage());  // if request read timeout, exception will be here and  method ```NACOS_RESTTEMPLATE.get``` will take at least ```readTimeoutMs``` ms.
            } catch (Exception ex) {

        } while (System.currentTimeMillis() <= endTime); // And this condition will fail with a high probability.
        
        LOGGER.error("no available server"); // So the error message is not exactly in most cases. It's a misleading message.
        throw new ConnectException("no available server");
    }

```

Reply:
And what said is not ture, and maybe it's just a timeout case.
> When code run in there, means all server nodes has retried